drivers/can/can_sam.c

```diff
@@ -101,7 +102,13 @@ static int can_sam_init(const struct device *dev)
 		return ret;
 	}
 
-	ret = can_mcan_configure_mram(dev, 0U, sam_cfg->mram);
+	/* get actual message ram base address */
+	uint32_t mrba = sam_cfg->mram & 0xFFFF0000;
+
+	/* keep lower 16bit; update DMA Base Address */
+	sys_write32(sam_cfg->dma_base, (sys_read32(sam_cfg->dma_base) & 0x0000FFFF) | mrba);
```

Comment:
I think the sys_read is not necessary.

Reply:
`sys_read()` seems to be just a cast from address to pointer value. from my perspective, it doesnt make any difference in using `sys_read()` or casting this here to pointer (which is needed anyway.

Reply:
This has the order of the arguments to `sys_write32()` reversed. Fix here: https://github.com/zephyrproject-rtos/zephyr/pull/80752